pywinauto/linux/atspi_objects.py

```diff
@@ -491,7 +502,67 @@ def get_iface_func(self, func_name):
             return None
 
 
+class GLIB(IATSPI):
+
+    """Python wrapper around C functions from GLib library"""
+
+    LIB = "libglib-2.0.so"
+
+
+_GHFunc = CFUNCTYPE(c_void_p, c_char_p, c_char_p)
+
+_glib = cdll.LoadLibrary(GLIB.LIB)
+
+_g_str_hash = _glib.g_str_hash
+_g_str_hash.restype = c_uint
+_g_str_hash.argtypes = [c_char_p]
+_GStrHashFunc = CFUNCTYPE(c_uint, c_char_p)
+
+_g_str_equal = _glib.g_str_equal
+_g_str_equal.restype = c_bool
+_g_str_equal.argtypes = [c_char_p, c_char_p]
+_GStrEqualFunc = CFUNCTYPE(c_bool, c_char_p, c_char_p)
+
+# Basic GHashTable constructor, to be used only with
+# string-based key-value pairs that are alloc/free by Python
+_g_hash_table_new = _glib.g_hash_table_new
+_g_hash_table_new.restype = c_void_p
+_g_hash_table_new.argtypes = [_GStrHashFunc, _GStrEqualFunc]
+
+_g_hash_table_foreach = _glib.g_hash_table_foreach
+_g_hash_table_foreach.restype = None
+_g_hash_table_foreach.argtypes = [c_void_p, _GHFunc, c_void_p]
+
+_g_hash_table_destroy = _glib.g_hash_table_destroy
+_g_hash_table_destroy.restype = None
+_g_hash_table_destroy.argtypes = [c_void_p]
+
+_g_hash_table_insert = _glib.g_hash_table_insert
+_g_hash_table_insert.restype = c_bool
+_g_hash_table_insert.argtypes = [c_void_p, c_void_p, c_void_p]
```

Comment:
I was following the same pattern we have in ```win32functions```, just a set of interfaces to native functions, but it might look good if grouped logically under a class. However, not sure about routing everything through a method similar to ```get_iface_func``` as there is no ambibguity in Glib function names like in ATSPI interfaces.

Reply:
BTW, hope you noticed, it's strictly specialized for strings and quite far from a full fledged python wrapper arount g_hash_table.